src/DataSource/DoctrineDataSource.php

```diff
@@ -343,4 +349,27 @@ public function getPlaceholder()
 		return $this->placeholder++;
 	}
 
+	/**
+	 * @param string $aggregation_type
+	 * @param string $column
+	 * @return mixed
+	 */
+	public function addAggregationColumn($aggregation_type, $column)
+	{
+		$this->aggregations[$column] = $aggregation_type;
+	}
+
+	/**
+	 * get aggregation row
+	 * @return array
+	 */
+	public function getAggregationData()
+	{
+		$this->data_source->resetDQLPart('select');
+		$root_alias = $this->data_source->getRootAliases()[0];
+		foreach ($this->aggregations as $column => $aggregation_type) {
+			$query = $this->data_source->select("$aggregation_type($root_alias.$column) AS $column");
```

Comment:
You should use `$this->checkAliases($column)` so if there is alias already mentioned, it still works, otherwise good job!